lib/foreman/password_hash.rb

```diff
@@ -9,12 +9,32 @@
 #
 module Foreman
   class PasswordHash
+    class PBKDF2Implementation
+      SIZE = 39
+      ITERATIONS = 4096
```

Comment:
Cost should be also configurable, like for bcrypt. The safe default value should be probably 100,000 iterations.